File: src/main/java/ski/mashiro/dto/Result.java
package ski.mashiro.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author MashiroT
 */
@Data
@NoArgsConstructor
public class Result<T>{
    private int code;
    private T data;
    private String msg;

    public Result(int code, T data, String msg) {
        this.code = code;
        this.data = data;
        this.msg = msg;
    }

    public static <T> Result<T> success(int code, T data) {
        return new Result<>(code, data, null);
    }

    public static <T> Result<T> failed(int code, String msg) {
        return new Result<>(code, null, msg);
    }
}

File: src/main/kotlin/ski/mashiro/common/Result.kt
package ski.mashiro.common

/**
 * @param code 状态码
 * @param data 返回数据
 * @param msg  响应消息
 * @param <T>  数据类型
 * @author MashiroT
 * @since 2022-06-28
</T> */
data class Result<T>(val code: Int, val data: T, val msg: String?) {
    companion object {
        fun <T> success(code: Int, data: T): Result<T> {
            return Result(code, data, null)
        }

        fun <T> failed(code: Int, msg: String?): Result<T?> {
            return Result(code, null, msg)
        }
    }
}
